lib/src/components/text_input/text_input.dart

```diff
@@ -0,0 +1,446 @@
+                  })
+              : null,
+          onExit: !widget.disabled
+              ? (_) => setState(() {
+                    _hovered = false;
+                  })
+              : null,
+          child: TextFormField(
+            enabled: !widget.disabled,
+            key: _key,
+            controller: _controller,
+            inputFormatters: widget.inputFormatters,
+            validator: (val) {
+              setState(() {
+                _errorText = widget.validator?.call(val);
+              });
+              return _errorText;
+            },
+            textAlignVertical: TextAlignVertical.center,
+            onChanged: widget.onChange,
+            style: _baseTextStyle,
+            cursorErrorColor: _colors.error,
+            decoration: InputDecoration(
+              isDense: true,
+              contentPadding: _contentPadding,
+              filled: true,
+              prefixIcon: _prefix,
+              prefixIconConstraints: widget.prefixText != null ? _affixConstraints : null,
+              suffixIcon: _suffix,
+              suffixIconConstraints: _affixConstraints,
+              focusColor: _backgroundColor,
+              hoverColor: _backgroundColor,
+              fillColor: _backgroundColor,
+              enabledBorder: _baseBorder,
+              disabledBorder: _baseBorder,
+              focusedBorder: _focusedBorder,
+              focusedErrorBorder: _errorBorder,
+              errorBorder: widget.disabled ? _baseBorder : _errorBorder,
+              hintText: widget.placeholder,
+              errorText: _errorText,
+              hintStyle: _baseTextStyle,
+              errorStyle: const TextStyle(height: 0.001, color: Colors.transparent),
+            ),
+          ),
+        ),
+        _HintText(
+          disabled: widget.disabled,
+          rounded: widget.rounded,
+          hintText: widget.hintText,
+          errorText: _errorText,
+        ),
+      ],
+    );
+  }
+}
+
+class _Label extends StatelessWidget {
+  const _Label({
+    required this.label,
+    required this.requirementLevel,
+    required this.disabled,
+  });
+
+  final String label;
+  final ZetaFormFieldRequirement requirementLevel;
+  final bool disabled;
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = Zeta.of(context).colors;
+    const textStyle = ZetaTextStyles.bodyMedium;
+
+    Widget? requirementWidget;
+
+    if (requirementLevel == ZetaFormFieldRequirement.optional) {
+      requirementWidget = Text(
+        '(optional)', // TODO(UX-1003): needs localizing.
+        style: textStyle.copyWith(color: disabled ? colors.textDisabled : colors.textSubtle),
+      );
+    } else if (requirementLevel == ZetaFormFieldRequirement.mandatory) {
+      requirementWidget = Text(
+        '*',
+        style: ZetaTextStyles.labelIndicator.copyWith(
+          color: disabled ? colors.textDisabled : colors.error, // TODO(mikecoomber): change to textNegative when added
```

Comment:
todo

Reply:
this color isn't in Zeta yet, hence the todo